### PR TITLE
fix: error handling in opening of external file.

### PR DESCRIPTION
**Pull request by imdeep2905**
#### Description of Change
Added error handling in opening of external file.

#### Checklist

- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#New-File-Name-guidelines)
- [x] Added tests and example, test must pass
- [x] Added documentation so that the program is self-explanatory and educational - [Doxygen guidelines](https://www.doxygen.nl/manual/docblocks.html)
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes: 

<a href="https://gitpod.io/#https://github.com/TheAlgorithms/C-Plus-Plus/pull/1043"></a>

### CODE REVIEW

**Comment by imdeep2905** [comment]
I think some unwanted commits are here. Let me fix them so I only modify ```neural_network.cpp```.

File: DIRECTORY.md [hunk 1]
```diff
@@ -251,4 +252,4 @@
 ## Strings
   * [Brute Force String Searching](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/strings/brute_force_string_searching.cpp)
   * [Knuth Morris Pratt](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/strings/knuth_morris_pratt.cpp)
-  * [Rabin Karp](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/strings/rabin_karp.cpp)
+  * [Rabin Karp](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/strings/rabin_karp.cpp)
```
Comment (Panquesito7) [inline DIRECTORY.md:255]: `DIRECTORY.md` is still being modified.
IMO, this file shouldn't be modified here.

Comment (imdeep2905) [reply DIRECTORY.md:255]: Yeah it was accidentally included in this PR. It looks like when I updated my fork I got commits which you guys reverted.

Forking repo again after this PR ✔️

**Comment by imdeep2905** [comment]
I am closing this PR opening new one.

Looks like something is wrong with my fork.

**Pull request closed without merging**